tests/TestRvsdgs.hpp

```diff
@@ -2537,4 +2537,136 @@ class VariadicFunctionTest2 final : public RvsdgTest
   rvsdg::Node * AllocaNode_ = {};
 };
 
+/**
+ * This class sets up an RVSDG representing the following code:
+ *
+ * \code{.c}
+ *   char* getPtr();
+ *
+ *   extern int global;
+ *   extern short globalShort;
+ *   extern int array[10];
+ *
+ *   void func(int** p) {
+ *     int alloca1, alloca2;
+ *
+ *     // Calculate the same offset in two ways
+ *     int* q = *p + 2;
+ *     int* qPlus2 = *p + 4;
+ *     int* qAgain = qPlus2 - 2;
+ *
+ *     // Different offsets into storage instances
+ *     int* arr1 = array + 1;
+ *     int* arr2 = array + 2;
+ *     int* arr3 = array + 3;
+ *     int* arrUnknown = array + global;
+ *
+ *     // Make alloca2 "escape" the function
+ *     *p = &alloca2;
+ *
+ *     char* bytePtr = getPtr();
+ *     char* bytePtrPlus2 = bytePtr + 2;
+ *
+ *     // All alias queries happen here
+ *   }
+ * \endcode
+ */
+class LocalAliasAnalysisTest1 final : public RvsdgTest
```

Comment:
This is not actionable right now, but ... TestRvsdgs has become too much of a place to collect all kinds of graphs. It is a justified place if the graphs here are subject to "multiple kinds of tests", but otherwise it is becoming quite unwieldy.

Reply:
Yes I agree, these graphs are only really used for one test each, so it does not really make sense to place them there